api/src/auth/utils/generate-provider-callback-url.ts

```diff
@@ -0,0 +1,15 @@
+import { Url } from '../../utils/url.js';
+import type { Request } from 'express';
+
+/**
+ * Generate provider callback URL from origin URL
+ *
+ * @param req Express request object
+ * @param providerName OAuth provider name
+ * @returns url
+ */
+export function generateProviderCallbackUrl(req: Request, providerName: string): string {
```

Comment:
Ideally this does not depend on `Request` and accepts the requires params it needs. This will improve its testability and dependance on request

Reply:
This function should default to using the PUBLIC_URL if the expected data (origin/protocal + domain) is not provided